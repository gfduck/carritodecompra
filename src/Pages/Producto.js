import React, { useEffect, useState } from "react";
import { useParams, useLocation } from "react-router-dom";
import { Jumbotron, Button } from "react-bootstrap";
import { connect } from "react-redux";
import { buscarunproducto } from "./../redux/actions/buscarUnProducto";
import Loading from "./../Components/Loading";
import { addToCart } from "../redux/actions/cart";

const Producto = ({ dispatch, isFetching, producto }) => {
  const { categoria, nombreProducto } = useParams();
  const [item, setItem] = useState({});
  let location = useLocation();

  useEffect(() => {
    if (location.item) {
      setItem(location.item);
    } else {
      buscarunproducto(dispatch, nombreProducto);
    }
  }, []);

  useEffect(() => {
    if (location.item) {
      setItem(location.item);
    } else {
      setItem(producto[0]);
    }
  }, [producto]);
  return (
    <>
      {isFetching && <Loading />}
      {item?.nombre && (
        <Jumbotron
          style={{
            background: "#ebebeb",
            paddingTop: "20px",
            paddingBottom: "5px",
          }}
        >
          <h1 className="text-center" data-testid="producto">
            {item.nombre}
          </h1>
          <img
            alt="fotoProducto"
            src={item.imageUrl}
            className="img-fluid mx-auto d-block"
            style={{ width: "20vw" }}
          />
          <p className="text-center mt-2 mb-2">{item.descripcion}</p>
          <p>
            <Button
              className="mx-auto d-block"
              variant="primary"
              onClick={() =>
                dispatch(
                  addToCart({
                    ...item,
                    subtotal: item.precio,
                    cantidad: 1,
                  })
                )
              }
            >
              Agregar
            </Button>
          </p>
        </Jumbotron>
      )}
    </>
  );
};
const mapStateToProps = (state) => {
  return {
    producto: state.buscarUnProducto.data,
    isFetching: state.buscarUnProducto.isFetching,
  };
};
export default connect(mapStateToProps)(Producto);
