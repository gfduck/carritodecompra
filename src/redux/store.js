import { createStore, applyMiddleware } from "redux";
import rootReducer from "./reducers"; // hace referencia al file index.js
import thunk from "redux-thunk";
//, applyMiddleware(thunk)

// import { createStore, applyMiddleware } from "redux";
// import rootReducer from "./reducers"; // hace referencia al file index.js
// import thunk from "redux-thunk";
// const store = createStore(rootReducer, applyMiddleware(thunk));
// export default store;

// const store = createStore(rootReducer);
const store = createStore(rootReducer, applyMiddleware(thunk));
export default store;
