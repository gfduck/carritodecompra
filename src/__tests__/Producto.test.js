import React from "react";
import Producto from "./../Pages/Producto";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import store from "./../redux/store";

import { Provider } from "react-redux";

configure({ adapter: new Adapter() });
describe("test producto", () => {
  test("debe mostrar componente producto", () => {
    const wrapper = shallow(
      <Provider store={store}>
        <Producto />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
