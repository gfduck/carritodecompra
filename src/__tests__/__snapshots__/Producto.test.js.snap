// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test producto debe mostrar componente producto 1`] = `ShallowWrapper {}`;
